src/services/audius-api-client/ResponseAdapter.ts

```diff
@@ -118,3 +127,74 @@ export const makeTrack = (track: APITrack): UserTrackMetadata | undefined => {
 
   return marshalled
 }
+
+export const makePlaylist = (
+  playlist: APIPlaylist
+): UserCollection | undefined => {
+  const decodedPlaylistId = decodeHashId(playlist.id)
+  const decodedOwnerId = decodeHashId(playlist.user_id)
+  const user = makeUser(playlist.user)
+  if (!decodedPlaylistId || !decodedOwnerId || !user) {
+    return undefined
+  }
+
+  const saves = playlist.followee_favorites
+    .map(makeFavorite)
+    .filter(removeNullable)
+
+  const reposts = playlist.followee_reposts
+    .map(makeRepost)
+    .filter(removeNullable)
+
+  const playlistContents = {
+    track_ids: playlist.added_timestamps.map(ts => ({
+      track: decodeHashId(ts.track_id),
+      time: ts.timestamp
+    }))
+  }
+
+  const tracks = playlist.tracks.map(track => makeTrack(track))
+
+  const marshalled = {
+    ...playlist,
+    user,
+    tracks,
+    playlist_id: decodedPlaylistId,
+    playlist_owner_id: decodedOwnerId,
+    followee_saves: saves,
+    followee_reposts: reposts,
+    save_count: playlist.favorite_count,
+    playlist_contents: playlistContents,
+
+    // Fields to prune
+    id: undefined,
+    user_id: undefined,
+    followee_favorites: undefined,
+    artwork: undefined,
+    favorite_count: undefined,
+    added_timestamps: undefined
+  }
+
+  delete marshalled.id
+  delete marshalled.user_id
+  delete marshalled.followee_favorites
+  delete marshalled.artwork
+  delete marshalled.favorite_count
+  delete marshalled.added_timestamps
+
+  // @ts-ignore
+  return marshalled as UserCollection
```

Comment:
Why do we need this? The return type on the function signature doesn't do it?

Reply:
oh yeah i forgot to fix this too. this was broken because the Collection has OnChain and the Track does not in our models